lib/filewatcher/cycles.rb

```diff
@@ -35,10 +35,7 @@ def watching_cycle
 
     def trigger_changes(on_update = @on_update)
       thread = Thread.new do
-        changes = @every ? @changes : @changes.first(1)
-        changes.each do |filename, event|
-          on_update.call(filename, event)
-        end
+        on_update.call(@changes.dup)
```

Comment:
Why do you think that `dup` is required?

Reply:
Hm I made the change because of failing tests. Here's an example! (I retried this after rebasing)

```
  9) Filewatcher#watch when there are new files in subfolders is expected to eq [{"/path/to/filewatcher/spec/tmp/new_sub_folder"=>:updated, "/path/to/filewatcher/spec/tmp/new_sub_folder/file.txt"=>:created}]
     Failure/Error:
       expect(processed).to eq [
         { subfolder => :updated, watch_run.filename => :created }
       ]

       expected: [{"path/to/filewatcher/spec/tmp/new_sub_folder"=>:updated, "/path/to/filewatcher/spec/tmp/new_sub_folder/file.txt"=>:created}]
            got: [{}]

       (compared using ==)

       Diff:
       @@ -1,3 +1,2 @@
       -[{"/path/to/filewatcher/spec/tmp/new_sub_folder"=>:updated,
       -  "/path/to/filewatcher/spec/tmp/new_sub_folder/file.txt"=>:created}]
       +[{}]

     # ./spec/filewatcher_spec.rb:163:in `block (4 levels) in <top (required)>'
```

The `.dup` fixed a bunch of errors very similar to the one above, and each of them expected some non-empty hash but received an empty one.

Let me know if you have a better solution for this! I'm guessing that this used to work without the `dup` since the Enumerator was created before `@change.clear`. Now that we aren't iterating over the hash, it seems that it's being emptied before the test/user gets a chance to read from it.

Reply:
I think, for multi-threading things, like we have ones in specs, it's better to do `dup`, yes. Thank you.